api/src/main/java/org/openmrs/obs/handler/MediaHandler.java

```diff
@@ -61,7 +62,10 @@ public Obs getObs(Obs obs, String view) {
 				FileInputStream mediaStream = new FileInputStream(file);
 				ComplexData complexData = new ComplexData(originalFilename, mediaStream);
 				
-				complexData.setMimeType(OpenmrsUtil.getFileMimeType(file));
+				// Get the Mime Type and set it
+				String mimeType = getMimeTypeFromFile(file);
+				mimeType = mimeType != null ? mimeType : "application/octet-stream";
```

Comment:
If mimeType is null, shouldn't the default be like it was on line 64 before this change?

Reply:
I will replace it.

It will probably return the same thing, because if it can't read the type from the name, getMimeTypeFromFile() also probes the content type with Files.probeContentType(). I believe this also looks at the file extension. OpenmrsUtil.getFileMimeType() does that as well, but via a MimetypesFileTypeMap. However, from some stackoverflow posts it seems the behaviour of probeContentType() can be somehow buggy or confusing, so it might be better to do it as you suggested.